platform/parsers/elastic_query_dsl/pancake_sql_query_generation.go

```diff
@@ -37,10 +45,16 @@ func (p *pancakeSqlQueryGenerator) aliasedExprArrayToLiteralExpr(aliasedExprs []
 	return exprs
 }
 
-func (p *pancakeSqlQueryGenerator) generatePartitionBy(groupByColumns []model.AliasedExpr) []model.Expr {
+func (p *pancakeSqlQueryGenerator) generatePartitionBy(groupByColumns []model.GroupByExpr, useGroupByColumn bool) []model.Expr {
```

Comment:
I would suggest to rename old function to something like `generatePartitionByViaAliasExpr`  and add a new function `generatePartitionByViaGroupByExpr`